src/nvim/window.c

```diff
@@ -5452,7 +5458,9 @@ static void frame_setheight(frame_T *curfrp, int height)
   if (curfrp->fr_parent == NULL) {
     // topframe: can only change the command line
     if (height > ROWS_AVAIL) {
-      height = ROWS_AVAIL;
+      // If height is greater than the available space, try to create space for the frame by
+      // reducing 'cmdheight' if possible, while making sure `cmdheight` doesn't go below 1.
+      height = MIN(ROWS_AVAIL + (p_ch - 1), height);
```

Comment:
This PR is to automatically change the `'cmdheight'` value when the command line height is changed with `:resize` like operations.
However, it seems to me that these lines are for another purpose: reduce the command line height when maximizing a window (or making window height bigger).
This change is a breaking change and I wonder if it is expected.

Reply:
When `'cmdheight'` is greater than `1` this also causes to `wincmd _` to change `'cmdheight'` to `1`.

Reply:
@zeertzjq  Thank you for completing my comment!

Here's some screenshots to show what behavior is changed.
All the binaries I used are downloaded from the GitHub Releases.
- v0.6.1
![cmdheight-v061](https://user-images.githubusercontent.com/24771416/184540527-393ff23f-fea6-487c-b65e-33cd7f2c9ea7.gif)

- v0.7.0
![cmdheight-v070](https://user-images.githubusercontent.com/24771416/184540544-a6d8184d-190e-43aa-a3ec-0ac0a78b1196.gif)